Angular/src/app/components/filter-form/filter-form.component.ts

```diff
@@ -0,0 +1,44 @@
+import {Component, OnInit, Output, Input, EventEmitter} from '@angular/core';
+import {OptionChangedEvent} from "devextreme/ui/form";
+import {startViewDate, endViewDate} from "../../config";
+import {ICheckBoxOptions, IDateOptions, IFilterInputOptions} from "./config";
+
+@Component({
+  selector: 'app-filter-form',
+  templateUrl: './filter-form.component.html',
+  styleUrls: ['filter-form.component.css'],
+})
+
+export class FilterFormComponent implements OnInit {
+  @Output() onFilterValueChange = new EventEmitter<OptionChangedEvent>();
```

Comment:
Let's replace input/output for each filter value and replace it with universal one with FilterValues type:

```ts
export class FilterFormComponent implements OnInit {
  @Input() filterValues: FilterValues;
  @Output() filterValuesChange = new EventEmitter<FilterValues>();

  // other code.
}
```